### PR TITLE
feat!: bump eslint-config packages to latest

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### Overview

Bumps dependencies of our ESLint configuration package to their latest versions.

Corresponding monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/24182

### PR Checklist

- ~[ ] Related to designs:~
- [x] Related to JIRA ticket: WEB-1458
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Notes
You can read the full typescript-eslint docs at https://github.com/typescript-eslint/typescript-eslint. https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin has the full table of each rule's docs page link.

I don't know why the packages are not transitively brought in as dependencies in this updated package set. Do you? [[CircleCI failure](https://app.circleci.com/pipelines/github/Codecademy/client-modules/13431/workflows/6999b46b-72b3-418f-afad-7cee0a325a14/jobs/90919)]

The rollout plan will be to disable all the previously disabled rules on a per-repository basis. See linked PRs below.

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@4.0.2-alpha.3b392e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608b42b60f1aa600c983d397--gamut-preview.netlify.app>https://608b42b60f1aa600c983d397--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608b42b60f1aa600c983d397>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@4.0.2-alpha.dd70aa.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608ee11f60b9e5a45b75bcdf--gamut-preview.netlify.app>https://608ee11f60b9e5a45b75bcdf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608ee11f60b9e5a45b75bcdf>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@4.0.2-alpha.beb52e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608ee3ea60b9e5b6e875b9ff--gamut-preview.netlify.app>https://608ee3ea60b9e5b6e875b9ff--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608ee3ea60b9e5b6e875b9ff>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@4.0.2-alpha.7a2a5e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608ee820df726efff6d44135--gamut-preview.netlify.app>https://608ee820df726efff6d44135--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608ee820df726efff6d44135>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@4.0.2-alpha.60ceea.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608eea948f94e5d9c6cd683e--gamut-preview.netlify.app>https://608eea948f94e5d9c6cd683e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608eea948f94e5d9c6cd683e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@4.0.2-alpha.3fc90a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60919f2aee83a000c3dbbe17--gamut-preview.netlify.app>https://60919f2aee83a000c3dbbe17--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60919f2aee83a000c3dbbe17>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@4.0.2-alpha.6340ca.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6091a204b162041e8179f088--gamut-preview.netlify.app>https://6091a204b162041e8179f088--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6091a204b162041e8179f088>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@4.0.2-alpha.9790cc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6091a37cebae701107ed2664--gamut-preview.netlify.app>https://6091a37cebae701107ed2664--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6091a37cebae701107ed2664>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@4.0.2-alpha.047845.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6091b07dfacdce200d56b092--gamut-preview.netlify.app>https://6091b07dfacdce200d56b092--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6091b07dfacdce200d56b092>Deploy Logs</a></p>

File: .circleci/config.yml [hunk 1]
```diff
@@ -12,10 +12,10 @@ references:
       NODE_OPTIONS: '--max_old_space_size=8196'
       CIRCLE_TEST_REPORTS: /tmp/test-results
 
-  yarn_cache_key_1: &yarn_cache_key_1 v1-yarn-{{ arch}}-{{ checksum "yarn.lock" }}
+  yarn_cache_key_1: &yarn_cache_key_1 v2-yarn-{{ arch }}-{{ checksum "yarn.lock" }}
 
-  webpack_cache_key_1: &webpack_cache_key_1 v1-webpack-{{ arch}}-{{ checksum "yarn.lock" }}-{{ .Branch }}-{{ .Revision }}
-  webpack_cache_key_2: &webpack_cache_key_2 v1-webpack-{{ arch}}-{{ checksum "yarn.lock" }}-{{ .Branch }}
+  webpack_cache_key_1: &webpack_cache_key_1 v1-webpack-{{ arch }}-{{ checksum "yarn.lock" }}-{{ .Branch }}-{{ .Revision }}
+  webpack_cache_key_2: &webpack_cache_key_2 v1-webpack-{{ arch }}-{{ checksum "yarn.lock" }}-{{ .Branch }}
```
Comment (dreamwasp) [inline .circleci/config.yml:18]: 👌

**Review by dreamwasp — APPROVED** [review]

**Review by codecaaron — APPROVED** [review]
🚢

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@4.0.2-alpha.48838e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6093c3aa6765b49c98562271--gamut-preview.netlify.app>https://6093c3aa6765b49c98562271--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6093c3aa6765b49c98562271>Deploy Logs</a></p>

**Pull request merged**